mreg/api/v1/views.py

```diff
@@ -353,6 +355,33 @@ def get_queryset(self):
         qs = _host_prefetcher(super().get_queryset())
         return HostFilterSet(data=self.request.GET, queryset=qs).qs
 
+
+    def _create_ip(self, request: Request, host: Host, ip: str) -> None:
+        # We need an IP Address object to compare against the
+        # network and broadcast addresses of its network. 
+        try:
+            ipaddr = ipaddress.ip_address(ip)
+        except ValueError:
+            raise ValidationError(
+                {"ipaddress": "Invalid IP address."},
+            )
+
+        try:
+            network: Network = Network.objects.get(network__net_contains=ip)
+        except Network.DoesNotExist:
+            pass # network not in mreg
```

Comment:
Wouldn't this then allow anyone to create any DNS record as long as the network is outside of mreg? I think this needs to be restricted to prevent someone from registering www.google.com. Ie, if the network isn't in mreg *or* we're asking for the presumptive broadcast *or* we're asking the network address, we need to be superuser_or_admin or network_admin (or probably is_mreg_superuser_or_hostpolicy_admin + is_mreg_network_admin)?

Reply:
No, this just skips the check for broadcast and network adresses because we don't know the network boundary.

Normal access is governed by `IsGrantedNetGroupRegexPermission`, so that should not be an issue with neither the allowed name nor IPs.

Reply:
Right, we're assuming that this code path is never hit without the view's permissions checks already having filtered who can deal with non-mreg networks. 

This does however mean that we're effectively performing permissions checking in two places, so we could consider moving the test for broadcast and network address assignments to the permissions code itself? I mostly find it iffy to have to look multiple places to check for permissions.

Reply:
You're right. I'll look into implementing it as a permissions class.

Reply:
Tests are passing with this new approach, but we need some new tests for this scenario. There are so many conditionals that make it return early with `True` that it's likely we don't even hit the relevant checks in the tests.